engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -155,12 +155,10 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
           }
         }
 
-      case e: MethodReference => typeChildren(validationContext, node, current.popStack) { typedParams =>
-        TypeMethodReference(e.getName, current.stack, typedParams) match {
-          case Right(typingResult) => Valid(typingResult)
-          case Left(errorMsg) => if(strictMethodsChecking) invalid(errorMsg) else Valid(Unknown)
+      case e: MethodReference =>
+        current.stackHead.map(_ => extractMethodReference(e, validationContext, node, current.popStack)).getOrElse {
```

Comment:
you're popping stack twice... Let's maybe use .stackHead explicitly?

Reply:
Yep.. I've changed it.